docs/guides/form-validation.md

```diff
@@ -35,15 +35,15 @@ export default function Signup() {
 In this step, we'll define a server `action` in the same file as our `Signup` component. Note that the aim here is to provide a broad overview of the mechanics involved rather than digging deep into form validation rules or error object structures. We'll use rudimentary checks for the email and password to demonstrate the core concepts.
 
 ```tsx filename=app/routes/signup.tsx
-import type { ActionArgs } from "@remix-run/node"; // or cloudflare/deno
-import { json } from "@remix-run/node"; // or cloudflare/deno
-import { Form, redirect } from "@remix-run/react";
+import type { DataFunctionArgs } from "@remix-run/node"; // or cloudflare/deno
+import { json, redirect } from "@remix-run/node"; // or cloudflare/deno
+import { Form } from "@remix-run/react";
 
 export default function Signup() {
   // omitted for brevity
 }
 
-export async function action({ request }: ActionArgs) {
+export async function action({ request }: DataFunctionArgs) {
```

Comment:
Can we use `ActionFunctionArgs` here?  Technically loaders and actions share the same params and can use `DataFunctionArgs` - but that may not aleways be the case so we try to document them using the more specific versions.